File: examples/java-maven/src/main/java/airco/DecreaseTemperatureAction.java
package airco;

import org.jeasy.rules.api.Action;
import org.jeasy.rules.api.FactsMap;

public class DecreaseTemperatureAction implements Action<FactsMap> {

    static DecreaseTemperatureAction decreaseTemperature() {
        return new DecreaseTemperatureAction();
    }

    @Override
    public void execute(FactsMap facts){
        System.out.println("It is hot! cooling air..");
        Integer temperature = facts.get("temperature");
        facts.put("temperature", temperature - 1);
    }
}

File: easy-rules-tutorials/src/main/java/org/jeasy/rules/tutorials/airco/DecreaseTemperatureAction.kt
package org.jeasy.rules.tutorials.airco

import org.jeasy.rules.api.Action
import org.jeasy.rules.api.FactsMap

class DecreaseTemperatureAction : Action<FactsMap> {


    @Throws(Exception::class)
    override fun execute(facts: FactsMap) {
        System.out.println("It is hot! cooling air..")
        val temperature = facts.get("temperature") as Int
        facts.put("temperature", temperature - 1)
    }

    companion object {

        internal fun decreaseTemperature(): DecreaseTemperatureAction {
            return DecreaseTemperatureAction()
        }
    }
}
